webapp/channels/src/components/dialog_router/interactive_dialog_adapter.tsx

```diff
@@ -0,0 +1,844 @@
+        const submission: Record<string, unknown> = {};
+
+        if (!elements) {
+            return submission;
+        }
+
+        elements.forEach((element) => {
+            const value = values[element.name];
+
+            if (value === null || value === undefined) {
+                // Skip null/undefined values unless field is required
+                if (!element.optional && this.conversionContext.validateInputs) {
+                    this.logWarn('Required field has null/undefined value', {
+                        fieldName: element.name,
+                        fieldType: element.type,
+                        isOptional: element.optional,
+                    });
+                }
+                return;
+            }
+
+            // Type-safe value conversion with validation
+            switch (element.type) {
+            case 'text':
+            case 'textarea':
+                if (element.subtype === 'number') {
+                    // Handle numeric inputs
+                    const numValue = Number(value);
+                    submission[element.name] = isNaN(numValue) ? this.sanitizeString(value) : numValue;
+                } else {
+                    const stringValue = this.sanitizeString(value);
+
+                    // Validate length constraints
+                    if (this.conversionContext.validateInputs) {
+                        if (element.min_length !== undefined && stringValue.length < element.min_length) {
+                            this.logWarn('Field value too short', {
+                                fieldName: element.name,
+                                actualLength: stringValue.length,
+                                minLength: element.min_length,
+                            });
+                        }
+                        if (element.max_length !== undefined && stringValue.length > element.max_length) {
+                            this.logWarn('Field value too long', {
+                                fieldName: element.name,
+                                actualLength: stringValue.length,
+                                maxLength: element.max_length,
+                            });
+                        }
+                    }
+                    submission[element.name] = stringValue;
+                }
+                break;
+            case 'bool':
+                submission[element.name] = Boolean(value);
+                break;
+
+            case 'radio':
+                submission[element.name] = this.sanitizeString(value);
+                break;
+
+            case 'select':
+                if (element.data_source === 'users' || element.data_source === 'channels') {
+                    // Handle user/channel selects
+                    if (typeof value === 'object' && value !== null && 'value' in value) {
+                        submission[element.name] = this.sanitizeString((value as AppSelectOption).value);
```

Comment:
I understand what you are trying to do here, but this will probably break things, since the value of a select has a distinct meaning, so we shouldn't be changing it.

That being said, probably nobody is using values with potential to be sanitized in normal scenarios...